### DISCUSSION TITLE
Is there a way to extend options for built-in deploy/remove commands?

### DISCUSSION THREAD

**Discussion by coyoteecd** [serverless/serverless#10618] [Q&A] [answered]
In our stacks we use serverless-s3-sync plugin to copy some files to S3 buckets after stack deployment The plugin has an option that can be passed together with the sls deploy/remove commands, as documented here: https://github.com/k1LoW/serverless-s3-sync#usage. We use the option to disable the plugin during sls remove.

With v3, support for unrecognized options is going away, so the warning that the plugin gives will become an error.
I know that a plugin can declare custom options for its commands. 

@medikoo @pgrzesik: is there a way for a plugin to declare additional options for the built-in deploy/remove command?

**Comment by medikoo** [comment, marked as answer]
> @medikoo @pgrzesik: is there a way for a plugin to declare additional options for the built-in deploy/remove command?

@coyoteecd yes, e.g. see how it's done here: https://github.com/serverless/serverless-plugin-typescript/blob/8914e65ccfc16272142f2c4134f27bbb43f8c3ba/src/index.ts#L25-L38 (it adds an option for built-in `invoke local` command)

**Reply by coyoteecd** [reply]
Worked great, thank you!
